### DISCUSSION TITLE
Does training instances expand during model training?

### DISCUSSION THREAD

**Discussion by nehalAggarwal** [online-ml/river#490] [Q&A]
Once the model gets trained on the instance, does the instance get discarded, or is it stored somewhere in the model and used to train the model again with the new instance (like an expanding window)? 

For example, if there are 10 samples, after learning the 1st sample, is 1st sample included while the model is trained on 2nd sample?  Similarly, for the 10th sample, are all the previous 9 samples included for training? If not, what happens to those samples?

**Comment by MaxHalford** [comment]
When you process an sample, it gets discarded. It's up to you if you want to store it aside and show it again to the model. The samples are not stored inside the model for further processing. This is the definition of "online".
